bots/make-checkout

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python2
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import os
+import subprocess
+import sys
+
+sys.dont_write_bytecode = True
+
+# Check out the given ref and if necessary overlay the bots
+# directory on top of it as expected on non-master branches
+
+BOTS = os.path.dirname(__file__)
+BASE = os.path.normpath(os.path.join(BOTS, ".."))
+
+def main():
+    parser = argparse.ArgumentParser(description="Ensure necessary issue exists for translations")
+    parser.add_argument("-v", "--verbose", action="store_true", help="Verbose output")
+    parser.add_argument("--base", nargs='?', help="Base branch that revision is for")
+    parser.add_argument("ref", help="The git ref to fetch")
+    parser.add_argument("revision", default="FETCH_HEAD", nargs='?',
```

Comment:
We don't use this ourselves, and it's a bit confusing to accept two git refspecs as positional arguments. This could be an actual `--revision` option.

Reply:
We do use this. See tests-scan. Basically we fetch a ref, and then check it out. The order makes sense as it's the order of the tasks, no?

Reply:
Ah, so you are concerned that by making it an option the order would be reversed. Fair enough, not important enough to dwell on this.